addons/snailmail_account/wizard/account_move_send.py

```diff
@@ -27,18 +29,30 @@ def _compute_checkbox_send_by_post(self):
             wizard.checkbox_send_by_post = not wizard.send_by_post_warning_message \
                 and wizard.company_id.invoice_is_snailmail
 
-    @api.depends('mode')
+    @api.depends('mode', 'checkbox_send_by_post')
     def _compute_send_by_post_warning_message(self):
         for wizard in self:
             display_messages = []
             if wizard.enable_send_by_post:
-                wrong_address_partners = wizard.move_ids.partner_id\
-                    .filtered(lambda x: not self.env['snailmail.letter']._is_valid_address(x))
-                if wrong_address_partners:
-                    display_messages.append(_("The following customers don't have a valid address: "))
-                    display_messages.append(", ".join(wrong_address_partners.mapped('display_name')))
+                invoice_without_valid_address = wizard.move_ids.filtered(
+                    lambda x: not self.env['snailmail.letter']._is_valid_address(x.partner_id))
+
+                if invoice_without_valid_address and wizard.checkbox_send_by_post:
+                    display_messages.append(_(
+                        "The partners on the following invoices have no valid address, "
+                        "so those invoices will not be sent: "
+                    ))
+                    display_messages.append(", ".join(invoice_without_valid_address.mapped('name')))
             wizard.send_by_post_warning_message = "".join(display_messages) if display_messages else None
 
+    @api.depends('move_ids')
+    def _compute_send_by_post_extra_fields(self):
+        for wizard in self:
+            partner_with_valid_address = wizard.move_ids.partner_id.filtered(
+                lambda x: self.env['snailmail.letter']._is_valid_address(x))
+            wizard.send_by_post_cost = len(partner_with_valid_address)
+            wizard.send_by_post_readonly = not partner_with_valid_address
```

Comment:
again, it's weird `send_by_post_readonly` is not computed at the same time as `send_by_post_warning_message`